### PR TITLE
Separate models that can be a top level resource and property of another resource. Fix path parameter name for app service certificate order

### PR DESCRIPTION
**Pull request by naveedaz**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [X] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [X] If applicable, the PR references the bug/issue that it fixes.
- [X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [X] My spec meets the review criteria:
  - [X] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [X] Validation errors from the [Linter extension for VS Code](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/linter.md) have all been fixed for this spec. (**Note:** for large, previously checked in specs, there will likely be many errors shown. Please contact our team so we can set a timeframe for fixing these errors if your PR is not going to address them).
  - [X] The spec follows the patterns described in the [Swagger good patterns](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-good-patterns.md) document unless the service API makes this impossible.

### CODE REVIEW

**Comment by naveedaz** [comment]
@anuchandy The CI keeps failing on changes unrelated to this PR. Can you please look into it?

**Comment by anuchandy** [comment]
@naveedaz CI issue is fixed and checks are passing now.

File: arm-web/2015-08-01/swagger/AppServiceCertificateOrders.json
Comment (anuchandy) [inline arm-web/2015-08-01/swagger/AppServiceCertificateOrders.json]: 1. It seems we renamed 'AppServiceCertificate' to 'AppServiceCertificateResource', any specific reason for the new name?
2. 'AppServiceCertificate' model is not used anywhere right? if so please remove it.

File: arm-web/2015-08-01/swagger/AppServiceCertificateOrders.json
Comment (anuchandy) [inline arm-web/2015-08-01/swagger/AppServiceCertificateOrders.json]: what is 'Csm' (can we expand it?) or does just "Key Vault resource Id." works?

File: arm-web/2016-09-01/swagger/AppServicePlans.json
Comment (anuchandy) [inline arm-web/2016-09-01/swagger/AppServicePlans.json]: Confirming - Is this intentional? removal of "Resource" hence making SiteConfig not an ARM tracked resource anymore?

File: arm-web/2016-09-01/swagger/AppServicePlans.json
Comment (anuchandy) [inline arm-web/2016-09-01/swagger/AppServicePlans.json]: Confirming - Is this intentional? removal of "Resource" hence making VnetInfo not an ARM tracked resource anymore?

File: arm-web/2016-08-01/swagger/WebApps.json
Comment (anuchandy) [inline arm-web/2016-08-01/swagger/WebApps.json]: Is this intentional? changing VnetInfo to non-tracked resource?

File: arm-web/2016-08-01/swagger/WebApps.json
Comment (anuchandy) [inline arm-web/2016-08-01/swagger/WebApps.json]: Looks like the new model SiteConfigResource replaces this model, if that is the case can we remove this duplicate model?

File: arm-web/2016-09-01/swagger/AppServiceEnvironments.json
Comment (anuchandy) [inline arm-web/2016-09-01/swagger/AppServiceEnvironments.json]: remove this model if AppServiceEnvironmentResource replaces this

File: arm-web/2016-09-01/swagger/AppServiceEnvironments.json
Comment (anuchandy) [inline arm-web/2016-09-01/swagger/AppServiceEnvironments.json]: remove this model if WorkerPoolResource replaces this

**Review by anuchandy — CHANGES_REQUESTED** [review]
@naveedaz please take a look at the linter failures reported here https://travis-ci.org/Azure/azure-rest-api-specs/jobs/210689666

File: arm-web/2015-08-01/swagger/AppServiceCertificateOrders.json
Comment (naveedaz) [inline arm-web/2015-08-01/swagger/AppServiceCertificateOrders.json]: We generate the swagger from our REST API. The model AppServiceCertificate was used as a top-level resource and as a property in another resource. In swagger then the definition AppServiceCertificate had the ARM envelope properties (id,name,kind,location, sku etc.) even when a property of another resource. This was an issue as it did not represent the actual payload. The change here now has 2 models, AppServiceCertificate and AppServiceCertificateResource and both are used.

File: arm-web/2016-09-01/swagger/AppServicePlans.json
Comment (naveedaz) [inline arm-web/2016-09-01/swagger/AppServicePlans.json]: Yes this is intentional. Same explanation as AppServiceCertificate vs AppServiceCertificateResource.

Comment (naveedaz) [inline arm-web/2016-09-01/swagger/AppServicePlans.json]: Both models are used.

File: arm-web/2016-09-01/swagger/AppServicePlans.json
Comment (naveedaz) [inline arm-web/2016-09-01/swagger/AppServicePlans.json]: Yes this is intentional. Same explanation as AppServiceCertificate vs AppServiceCertificateResource. Both models are used.

File: arm-web/2016-08-01/swagger/WebApps.json
Comment (naveedaz) [inline arm-web/2016-08-01/swagger/WebApps.json]: Same explanation as AppServiceCertificate vs AppServiceCertificateResource. Both models are used.

File: arm-web/2016-09-01/swagger/AppServiceEnvironments.json
Comment (naveedaz) [inline arm-web/2016-09-01/swagger/AppServiceEnvironments.json]: Yes this is intentional. Same explanation as AppServiceCertificate vs AppServiceCertificateResource. Both models are used.

File: arm-web/2016-09-01/swagger/AppServiceEnvironments.json
Comment (naveedaz) [inline arm-web/2016-09-01/swagger/AppServiceEnvironments.json]: Yes this is intentional. Same explanation as AppServiceCertificate vs AppServiceCertificateResource. Both models are used.

File: arm-web/2015-08-01/swagger/AppServiceCertificateOrders.json
Comment (anuchandy) [inline arm-web/2015-08-01/swagger/AppServiceCertificateOrders.json]: Thanks Navy. It seems we are duplicating properties in these two models, i just had a discussion with @amarzavery and confirmed that we could share properties here as:


```json
   "definitions": {
      "AppServiceCertificateResource": {
        "description": "Key Vault container ARM resource for a certificate that is purchased through Azure.",
        "type": "object",
        "allOf": [
          {
            "$ref": "#/definitions/Resource"
          }
        ],
        "properties": {
          "properties": {
            "$ref": "#/definitions/AppServiceCertificate",
            "x-ms-client-flatten": true
          }
        }
      },
      "AppServiceCertificate": {
        "description": "ARM envelope properties",
        "properties": {
          "keyVaultId": {
            "description": "Key Vault Csm resource Id.",
            "type": "string"
          },
          "keyVaultSecretName": {
            "description": "Key Vault secret name.",
            "type": "string"
          },
          "provisioningState": {
            "description": "Status of the Key Vault secret.",
            "enum": [
              "Initialized",
              "WaitingOnCertificateOrder",
              "Succeeded",
              "CertificateOrderFailed",
              "OperationNotPermittedOnKeyVault",
              "AzureServiceUnauthorizedToAccessKeyVault",
              "KeyVaultDoesNotExist",
              "KeyVaultSecretDoesNotExist",
              "UnknownError",
              "ExternalPrivateKey",
              "Unknown"
            ],
            "type": "string",
            "readOnly": true,
            "x-ms-enum": {
              "name": "KeyVaultSecretStatus",
              "modelAsString": false
            }
          }
        }
      }
    }
```

File: arm-web/2016-09-01/swagger/AppServiceEnvironments.json
Comment (anuchandy) [inline arm-web/2016-09-01/swagger/AppServiceEnvironments.json]: please check properties sharing described above can be applied for AppServiceEnvironment and AppServiceEnvironmentResource

File: arm-web/2016-09-01/swagger/AppServiceEnvironments.json
Comment (anuchandy) [inline arm-web/2016-09-01/swagger/AppServiceEnvironments.json]: please check properties sharing described above can be applied for WorkerPool and WorkerPoolResource

File: arm-web/2016-08-01/swagger/WebApps.json
Comment (anuchandy) [inline arm-web/2016-08-01/swagger/WebApps.json]: ok, please check properties sharing described above can be applied for SiteConfigResource and SiteConfig

File: arm-web/2016-08-01/swagger/WebApps.json
Comment (anuchandy) [inline arm-web/2016-08-01/swagger/WebApps.json]: fine, check properties can be shared as described above [AppServiceCertificateResource and AppServiceCertificate]

**Comment by anuchandy** [comment]
@naveedaz please take a look at the comments about avoiding duplicate properties, we can merge this after you address them.

## Notes
There are many errors and warnings in the web app swagger specs - https://travis-ci.org/Azure/azure-rest-api-specs/jobs/210689666. These errors are not related to the changes in this PR, but it would be great if your team can review them and plan to fix them in upcoming iterations, soon ARM team will be enforcing zero errors.

File: arm-web/2015-08-01/swagger/AppServiceCertificateOrders.json
Comment (jianghaolu) [inline arm-web/2015-08-01/swagger/AppServiceCertificateOrders.json]: @naveedaz Do you think this is achievable in your swashbuckle config?

**Comment by naveedaz** [comment]
@anuchandy, @amarzavery, @jianghaolu,
The CR comments are addressed. Please review and merge.

**Comment by anuchandy** [comment]
@naveedaz we are validating generated java SDK, after that we will merge this.

**Comment by jianghaolu** [comment]
It seems the only problem is type `WorkerPool` is not generated. Seems like an AutoRest issue? Other than that everything seems good to me. Thanks!

**Comment by jianghaolu** [comment]
Fix in AutoRest is submitted here: https://github.com/Azure/autorest/pull/2038. This PR LGTM.

**Review by anuchandy — APPROVED** [review]

**Comment by jianghaolu** [comment]
@naveedaz There's nothing on your side to do now - pending this PR: https://github.com/Azure/autorest/pull/2038. Without that this will break the SDKs.

**Pull request merged**
